packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/CameraPermissions.java

```diff
@@ -23,16 +24,18 @@ interface ResultCallback {
     void onResult(String errorCode, String errorDescription);
   }
 
+  private static final String TAG = "CameraPermissions";
   private static final int CAMERA_REQUEST_ID = 9796;
-  private boolean ongoing = false;
+  @VisibleForTesting boolean ongoing = false;
 
   void requestPermissions(
       Activity activity,
       PermissionsRegistry permissionsRegistry,
       boolean enableAudio,
       ResultCallback callback) {
     if (ongoing) {
-      callback.onResult("cameraPermission", "Camera permission request ongoing");
+      Log.e(TAG, "Another request is ongoing and multiple requests cannot be handled at once.");
```

Comment:
how would a Dart user handle this case?

Reply:
If I understand it correctly, this is a programming error that could be avoided by developers, which is different from other errors that depends on user interaction (`CameraAccessDenied`)

Reply:
On iOS side it's not even possible because the requests happen on the main thread. So duplicate requests would queue up one after another.

Reply:
This change creates a codepath that doesn't call the result callback, which violates the engine API.

> If I understand it correctly, this error is a programming mistake that could be avoided by developers

First they have to know about it though; even developer mistakes should be sent back over the channel to Dart. They can be converted to some kind of `Error` objects (rather than `Exception` objects; see https://api.dart.dev/stable/2.16.2/dart-core/Error-class.html).

Reply:
Gotcha, good to learn. I think i'm fine with sending this to dart as an Error. 


Do we expect developers to catch these errors? Something like: 
```
try {
  controller.initialize()
} catch (CameraException e) {
  // handle exceptions like before 
} catch (CameraError e) {
  if (e.code == "DuplicateRequest") {
    print("This should not have happened. There is a bug in camera usage and need to fix it.")
  }
}
```

Reply:
No, the documentation for `Error` explicitly says developers should not catch them.

People may, but they shouldn't. The important thing is that we surface them as errors in the context of the Dart code, rather than silently swallowing serious programmer errors.

Reply:
("silently" in the sense that an Android log message may never be noticed; an `Error` will be.)

Reply:
Gotcha. thanks for the explain.